### PR TITLE
add limitBAMsortRAM to STAR and make sampleConfig write to file

### PR DESCRIPTION
**Pull request by DavyCats**

### CODE REVIEW

File: biopet.wdl [hunk 1]
```diff
@@ -60,6 +60,7 @@ task SampleConfig {
     String? preCommand
     String tool_jar
     Array[File]+ inputFiles
+    String stdoutFile
```
Comment (rhpvorderman) [inline biopet.wdl]: This is not necessary. Wdl does specify a stdout() function.
https://github.com/openwdl/wdl/blob/master/versions/draft-2/SPEC.md#file-stdout

File: gatk.wdl [hunk 2]
```diff
@@ -271,13 +272,13 @@ task SplitNCigarReads {
...
         -R ${ref_fasta} \
-        -O ${output_bam} # might have to be -o depending on GATK version \
+        -O ${output_bam} \
         -L ${sep=' -L ' intervals}
     }
 
     output {
         File bam = output_bam
-        File bam_index = output_bam + ".bai"
+        File bam_index = sub(output_bam, ".bam$", ".bai")
```
Comment (rhpvorderman) [inline gatk.wdl]: Use `\\.` instead of `.` to regex match `.` (Otherwise will match any character).

File: star.wdl [hunk 3]
```diff
@@ -12,6 +12,7 @@ task Star {
     String? outStd
     String? twopassMode
     Array[String]? outSAMattrRGline
+    Int? limitBAMsortRAM
```
Comment (rhpvorderman) [inline star.wdl]: Just checking: Did you make sure does not also accept floats?

Comment (DavyCats) [reply star.wdl]: The STAR manual says it should be an int.

Comment (rhpvorderman) [reply star.wdl]: :+1:

**Pull request merged**